library/Vanilla/DateFilterSchema.php

```diff
@@ -0,0 +1,200 @@
+<?php
+/**
+ * @copyright 2009-2017 Vanilla Forums Inc.
+ * @license GNU GPLv2
+ */
+
+namespace Vanilla;
+
+use DateTimeImmutable;
+use Garden\Schema\Invalid;
+use Garden\Schema\Schema;
+use Garden\Schema\ValidationField;
+use Garden\Schema\ValidationException;
+
+class DateFilterSchema extends Schema {
+
+    /** @var string Regular expression for matching a datetime. */
+    private $dateRegEx = '(?<date>\d{4}-\d{2}-\d{2})(?<time>[ T]\d{2}:\d{2}(:\d{2})?)?';
+
+    /** @var array Valid characters for opening an interval-notation range. */
+    private $rangeOpen = ['(', '['];
+
+    /** @var array Valid characters for closing an interval-notation range. */
+    private $rangeClose = [')', ']'];
+
+    /** @var array Valid operators for simple date comparisons. */
+    private $simpleOperators = ['=', '>', '<', '>=', '<='];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function __construct(array $schema = []) {
+        if (!empty($schema)) {
+            throw new \InvalidArgumentException(self::class.' does not support custom schemas.');
```

Comment:
Just declare the constructor with no parameter.
```php
/**
 * DateFilterSchema constructor.
 */
public function __construct() {
    parent::__construct();
}
```

Reply:
My intention here is to make sure that anyone attempting to pass a schema spec to this subclass immediately knows it isn't going to work, instead of having it silently fail to do what they were attempting.